### ISSUE TITLE
Test program does not work correctly on Windows 10 x86

### ISSUE THREAD

**Issue by DkmS1953** [AvaloniaUI/Avalonia#14864]
### Describe the bug

HelloWorld program works as expected everywhere (Linux, Windows) except Windows 10 x86 on Samsung tablet? The program window is drawn, but the text in the window is not.
Uwp programs on the same device work fine, test program on a similar framework - also works.

### To Reproduce

1. Create a project in Visual Studio 2022 using the "Avalonia C# Project" template.
2. Publish the App.Desktop project in the "Release/net8.0/win-x86" configuration.
3. Copy the resulting folder with files to a device on which the .NET 8.0 runtime system is pre-installed.
4. Run the App.Desktop.exe.

### Expected behavior

The program window with the expected title but without the text "Welcome to Avalonia!" will appear.

### Avalonia version

11.0.10

### OS

Windows

**Comment by timunie** [comment]
Font issue? Can you try to add a button or similar?

**Comment by thevortexcloud** [comment]
A screenshot would also probably help explain what it looks like.

**Comment by DkmS1953** [comment]
> Font issue? Can you try to add a button or similar?

Changed fonts, doesn't help.
The button is drawn, the text on the button is not.

**Comment by DkmS1953** [comment]
> A screenshot would also probably help explain what it looks like.
This is how it looks on Windows 10 x86
![image](https://github.com/AvaloniaUI/Avalonia/assets/25905429/fdb7c039-7975-4b9c-9986-e694da709ec0)

And this is how it looks on Ubuntu
![image](https://github.com/AvaloniaUI/Avalonia/assets/25905429/b0ffb7fc-79e5-484a-bc8a-60ed4c8193f9)

And this is how it looks on Windows 11 
![image](https://github.com/AvaloniaUI/Avalonia/assets/25905429/ec9f711e-a755-411d-aeaa-15c18422e7b7)

**Comment by timunie** [comment]
I still feel like it cannot access that font for some reason (sandboxed maybe?). I'd say you may want to try InterFont which has an avalonia package. Also it may help to see your sample uploaded here in case someone has a similar device to test.

**Comment by DkmS1953** [comment]
> I still feel like it cannot access that font for some reason (sandboxed maybe?). I'd say you may want to try InterFont which has an avalonia package. Also it may help to see your sample uploaded here in case someone has a similar device to test.

There's no sandboxing, just a normal Windows 10 installation on a Samsung tablet. So yeah, it's old, but that's no reason not to work on it. UWP and WinUI programs work quite well. Fonts are all in place, the latest build, which is in the screenshot, uses Courier New, which is everywhere.
The InterFont font doesn't save the day.

**Comment by timunie** [comment]
UWP is sandboxed iirc. Question would more be if WPF works. 

I don't say it should not work but I can't try on my own so I can only guess. Something must be how we handle fonts or text processing.

**Comment by DkmS1953** [comment]
> Also it may help to see your sample uploaded here in case someone has a similar device to test.

[AvaHello solution.7z.zip](https://github.com/AvaloniaUI/Avalonia/files/14546551/AvaHello.solution.7z.zip)

**Comment by DkmS1953** [comment]
> Question would more be if WPF works.

WPF on .NET 8 works fine.

**Comment by thevortexcloud** [comment]
Out of curiosity, is the Samsung tablet's display fairly small? I am thinking (albeit this is a somewhat baseless guess) this might be scaling/font size related. The fonts are also noticeably different in your screenshot. EG the Linux one is missing serifs.

Are you also using actual 32 bit windows? If so this might also be a dependency issue in a native library. I doubt many people test 32bit windows these days.

**Comment by DkmS1953** [comment]
> Out of curiosity, is the Samsung tablet's display fairly small

He has a normal monitor, 10", 1366*768, 125% scaling enabled (my eyesight is no longer eagle-eyed).
Parameters:
Processor	Intel(R) Atom(TM) CPU Z2760  @ 1.80GHz   1.80 GHz
Installed RAM	2,00 GB
System type	32-bit operating system, x86-based processor
Pen and touch	Pen and touch support with 5 touch points
Edition	Windows 10 Pro
Version	22H2
OS build	19045.4123
Experience	Windows Feature Experience Pack 120.2212.3030.0

There are no programs that can affect the text display in any way:
![image](https://github.com/AvaloniaUI/Avalonia/assets/25905429/da84f4fc-8d5e-4875-9562-87cf7188bf4e)

**Comment by DkmS1953** [comment]
Looks like it's not fated to run Avalonia software on my tablet.....

**Comment by WhereTheTimeWent** [comment]
I can reproduce this on Windows 10 Enterprise LTSC 2019 (1809) x64.
The text of TextBlock and TextBox doesn't show.
Changing FontFamily to Calibri (a font that definitely exists in this Win 10 version) and Foreground (of a TextBlock) to black doesn't change that.
Switching to dark mode also doesn't make a difference.
I use Avalonia 11.3.4 and .NET 9 (the runtime for 9.0.8 is installed). I tried publishing from Visual Studio, and just using the bin folder's debug files.
The project was created with the latest Visual Studio templates (Avalonia .NET App).

What I expect when starting an empty project:
<img width="1426" height="843" alt="Image" src="https://github.com/user-attachments/assets/90a88818-93e7-4795-bbba-a8370a9006e5" />

What I get on Enterprise LTSC 2019 (same .exe/build files):
<img width="1426" height="843" alt="Image" src="https://github.com/user-attachments/assets/4cd3be42-cba8-43c4-8b7e-d5fa70ebd62d" />


Steps to reproduce:
- Create Avalonia .NET App in Visual Studio 2022
- Build it
- Copy the bin\Debug folder to a Windows 10 Enterprise LTSC 2019 installation
- Launch the .exe

**Comment by lextm** [comment]
@WhereTheTimeWent Windows 10 version 1809 reached its end of life on [May 11, 2021](https://learn.microsoft.com/en-us/lifecycle/products/windows-10-enterprise-and-education), more than 4 years ago. And anyway, Windows 10 in general will deprecate on Oct 14, 2025.

So, even if this can be reproduced, I think it is not worth the while to fix (unless you observe the same behavior on a Windows 11 release that is still in active lifecycle).

**Comment by thevortexcloud** [comment]
> Windows 10 in general will deprecate on Oct 14, 2025.

This is true for regular consumer versions. Windows 10 however will be supported for _many_ more years. EG  Some LTSC versions will be supported until 2029 and some of the IoT versions will be supported until 2032.

https://learn.microsoft.com/en-us/windows/whats-new/ltsc/overview
https://learn.microsoft.com/en-us/windows/iot/iot-enterprise/whats-new/release-history#windows-iot-enterprise-ltsc

It's also likely Windows 10 will end up in a Windows XP situation where Microsoft keeps providing updates to those willing to pay for them for the next ~20 years.

That said: Even on regular consumer versions you can also pay for updates or opt into some shenanigans to get more updates beyond the official EOL. Microsoft just really wants people to move onto Windows 11 so they wont talk about this much.

**Comment by timunie** [comment]
If anyone wants to provide a PR that fixes the issue, feel free to do. We can't reproduce it on our end, so we can only fix it "by accident".

**Comment by WhereTheTimeWent** [comment]
It works as expected on Windows 10 22H2.

~~It doesn't work on 1809.~~ Both Enterprise LTSC 2019 and Server 2019 have support until 2029-01-09.
I didn't test it on Server 2019, but I'm gonna do that later.

Windows Server 2019 is still wildly used for terminal servers. ~~Not gonna lie, I can't consider using Avalonia at work now, because we have a lot of Server 2019 at our customers.~~ 

Edit: It's not a general issue, it only happens with one of my installs.

**Comment by WhereTheTimeWent** [comment]
> I didn't test it on Server 2019, but I'm gonna do that later.

I can't reproduce it on Windows Server 2019, so we're good there. Seems to only affect Enterprise LTSC 2019 and whatever @DkmS1953 used.
I'm going to test it with a fresh installation of Enterprise LTSC 2019 tonight.

If someone is interested in logs, I'd be happy to provide them, if you tell me what logs you want and how you want them captured.

**Comment by WhereTheTimeWent** [comment]
> I'm going to test it with a fresh installation of Enterprise LTSC 2019 tonight.

Can't reproduce it on a fresh install of Enterprise LTSC 2019.
